code/game/objects/structures/mortar.dm

```diff
@@ -234,51 +230,45 @@
 		to_chat(user, "<span class='warning'>[src]'s barrel is still steaming hot. Wait a few seconds and stop firing it.</span>")
 		return
 
-	playsound(loc, 'sound/items/ratchet.ogg', 25, 1)
-	user.visible_message("<span class='notice'>[user] starts undeploying [src].",
-	"<span class='notice'>You start undeploying [src].")
-	if(!do_after(user, 40, TRUE, src, BUSY_ICON_BUILD))
-		return
+	return ..()
 
-	user.visible_message("<span class='notice'>[user] undeploys [src].",
-	"<span class='notice'>You undeploy [src].")
-	playsound(loc, 'sound/items/deconstruct.ogg', 25, 1)
-	new /obj/item/mortar_kit(loc)
-	qdel(src)
+/obj/machinery/deployable/mortar/disassemble(mob/user)
+	. = ..()
+	SEND_SIGNAL(src, COMSIG_PARENT_QDELETING)
 
-/obj/structure/mortar/fixed
+/obj/machinery/deployable/mortar/fixed
 	desc = "A manual, crew-operated mortar system intended to rain down 80mm goodness on anything it's aimed at. Uses manual targetting dials. Insert round to fire. This one is bolted and welded into the ground."
-	fixed = TRUE
+	deploy_flags = DEPLOYED_NO_PICKUP
 
 //The portable mortar item
 /obj/item/mortar_kit
-	name = "\improper M402 mortar portable kit"
-	desc = "A manual, crew-operated mortar system intended to rain down 80mm goodness on anything it's aimed at. Needs to be set down first"
+	name = "\improper M402 mortar"
+	desc = "A manual, crew-operated mortar system intended to rain down 80mm goodness on anything it's aimed at. Needs to be set down first to fire."
 	icon = 'icons/Marine/mortar.dmi'
 	icon_state = "mortar_m402_carry"
+
+	deploy_icon_state = "mortar_m402"
+	deploy_time = 40
+
+	deploy_max_integrity = 200
+
+	deployed_type = /obj/machinery/deployable/mortar
+	flags_item = IS_DEPLOYABLE
+
 	resistance_flags = UNACIDABLE|INDESTRUCTIBLE
 	w_class = WEIGHT_CLASS_BULKY //No dumping this in most backpacks. Carry it, fatso
 
+/obj/item/mortar_kit/Initialize()
+	. = ..()
+	AddComponent(/datum/component/deployable_item, deploy_time, deployed_type)
```

Comment:
So let's use the mortar_kit as an exemple as what i think should be much better:

AddComponent(/datum/component/deployable_item, deploy_time = 4 SECONDS, deployed_type = /obj/machinery/deployable/mortar) and you can remove those vars from the item.

deploy max_integrity should be equal to max_integrity, no need for a second var here.

deploy_icon_state could be generated from the init(icon_state) without too much trouble